src/js/BookReader/utils/classes.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Exposes a function from one class (FromClass) to another (ToClass), in such a way
+ * that if the ToClass's method is overriden, the FromClass's method is also overriden.
+ * WARNING: This modifies FromClass' prototype! So FromClasses cannot be shared between
+ * different ToClasses.
+ * @template TFrom the class we'll be getting the method from
+ * @template TTo the class we'll be adding the method to
+ * @param {new () => TFrom} FromClass the class to get the method from
+ * @param {string} fromMethod the method's name in FromClass
+ * @param {function(TFrom): TTo} fromTransform how to get the TTo `this` to use when setting the method on TFrom
+ * @param {new () => TTo} ToClass the class to add the method to
+ * @param {string} toMethod the name of the method to add to TTo (likely will be equal to fromMethod)
+ * @param {function(TTo): TFrom} toTransform how to get the TFrom this to use when calling the new method
```

Comment:
This function is sure doing some heavy lifting. 💪 It makes me just love the power of JS even more ⚡️ 

There may be a better way to document this.  Some doc feedback - 
`TTo` and `TFrom` are getting a bit lost.
The function passes around scope, can we make that a bit clearer in this document block?

just brainstorming...
`toMethod the name of the method to add to TTo` => `toMethod - the method that gets updated with new/extended scope`

Reply:
It really is crazy; like this would be near impossible in most other languages (although to be fair it was kind of near impossible in JS, too :P ). The `@template` types are mostly an attempt to make this type check, but I didn't succeed; I'll see if I can get them to behave, and if not I'll drop them :+1:

Reply:
I think I'll rename this to `exposeOverrideable`; that seems a little clearer.

Reply:
Fixed; and got it fully to type-check, so auto-complete will even provide the _name of the method on FromClass_! I also pushed the `@template` type parameters to the bottom of the doc string; they're not really useful in this case, but needed for the autocompletion.